ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -1167,6 +1171,76 @@ def test_ds_configcheck_passwordstorage(self, modify_pwdstoragescheme):
                 assert "cn=config" in check["kw"]["items"]
                 assert error_msg in check["kw"]["msg"]
 
+    @pytest.fixture
+    def create_logfile(self):
+        """
+        This fixture creates an empty healthcheck.log file when
+        it is not present in /var/log/ipa/healthcheck/ directory so that
+        sosreport command collects the log file in the report generated
+        by the dry run and then deletes the file once the test is finished.
+        """
+        if not self.master.transport.file_exists(HEALTHCHECK_LOG):
+            self.master.run_command(
+                ["ipa-healthcheck", "--output-file", HEALTHCHECK_LOG]
+            )
+        yield
+        self.master.run_command(["rm", HEALTHCHECK_LOG])
+
+    def test_sosreport_includes_healthcheck(self, create_logfile):
+        """
+        This testcase checks that sosreport command
+        when run on IPA system with healthcheck installed
+        collects healthcheck.log file
+        """
+        caseid = "123456"
+        msg = "[plugin:ipa] collecting path '{}'".format(HEALTHCHECK_LOG)
+        cmd = self.master.run_command(
+            [
+                "sosreport",
+                "-o",
+                "ipa",
+                "--case-id",
+                caseid,
+                "--batch",
+                "-v",
+                "--build",
+            ]
+        )
+        assert msg in cmd.stdout_text
+
+    @pytest.fixture
+    def remove_healthcheck(self):
+        """
+        This fixture uninstalls healthcheck package on IPA
+        server and then installs.
+        """
+        tasks.uninstall_packages(self.master, HEALTHCHECK_PKG)
+        self.master.run_command(["rm", HEALTHCHECK_LOG])
+        yield
+        tasks.install_packages(self.master, HEALTHCHECK_PKG)
+
+    def test_sosreport_without_healthcheck_installed(self, remove_healthcheck):
+        """
+        This testcase checks that sosreport completes successfully
+        even if there is no healthcheck log file to collect
+        """
+        caseid = "123456"
+        if not self.master.transport.file_exists(HEALTHCHECK_LOG):
+            cmd = self.master.run_command(
+                [
+                    "sosreport",
+                    "-o",
+                    "ipa",
+                    "--case-id",
+                    caseid,
+                    "--batch",
+                    "-v",
+                    "--build",
+                ],
+                raiseonerr=False,
+            )
+        assert cmd.returncode == 0
```

Comment:
If you remove raiseonerr=False, you can also remove the assert (the test would raise an Exception if the command fails).